### ISSUE TITLE
failed to enable Telemetry V2 with Wasm runtime

### ISSUE THREAD

**Issue by richardwxn** [istio/istio#24115]
The issue is for latest master, proxy is not ready after enabling wasm

logs from proxy:
`2020-05-26T17:34:37.613280Z	warn	Envoy proxy is NOT ready: failed to get readiness stats: server.state is not yet updated: 
2020-05-26T17:34:39.712454Z	warn	Envoy proxy is NOT ready: failed to get readiness stats: Get "http://127.0.0.1:15000/stats?usedonly&filter=^(server.state|listener_manager.workers_started)": context deadline exceeded (Client.Timeout exceeded while awaiting headers)
2020-05-26T17:34:40.656862Z	critical	envoy backtrace	[bazel-out/k8-opt/bin/external/envoy/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:104] Caught Segmentation fault, suspect faulting address 0x0
2020-05-26T17:34:40.656912Z	critical	envoy backtrace	[bazel-out/k8-opt/bin/external/envoy/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:91] Backtrace (use tools/stack_decode.py to get line numbers):
2020-05-26T17:34:40.656915Z	critical	envoy backtrace	[bazel-out/k8-opt/bin/external/envoy/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:92] Envoy version: 4ec1c843fb74bb45d64f1b0202b200c4873eab88/1.14.1/Clean/RELEASE/BoringSSL
2020-05-26T17:34:41.124309Z	info	sds	resource:default connection is terminated: rpc error: code = Canceled desc = context canceled
2020-05-26T17:34:41.124395Z	error	sds	Remote side closed connection
2020-05-26T17:34:41.124435Z	info	sds	resource:ROOTCA connection is terminated: rpc error: code = Canceled desc = context canceled
2020-05-26T17:34:41.124440Z	error	sds	Remote side closed connection
2020-05-26T17:34:41.124456Z	info	sds	resource:default connection is terminated: rpc error: code = Canceled desc = context canceled
2020-05-26T17:34:41.124491Z	error	sds	Remote side closed connection
2020-05-26T17:34:41.124596Z	warn	Envoy proxy is NOT ready: failed to get readiness stats: Get "http://127.0.0.1:15000/stats?usedonly&filter=^(server.state|listener_manager.workers_started)": read tcp 127.0.0.1:35822->127.0.0.1:15000: read: connection reset by peer
2020-05-26T17:34:41.124620Z	info	transport: loopyWriter.run returning. connection error: desc = "transport is closing"
2020-05-26T17:34:42.712976Z	warn	Envoy proxy is NOT ready: failed to get readiness stats: Get "http://127.0.0.1:15000/stats?usedonly&filter=^(server.state|listener_manager.workers_started)": dial tcp 127.0.0.1:15000: connect: connection refused
2020-05-26T17:34:43.744720Z	error	Epoch 0 exited with error: signal: segmentation fault (core dumped)
2020-05-26T17:34:43.744778Z	info	No more active epochs, terminating`

**Comment by bianpengyuan** [comment]
Does this happens consistently in your deployment? We have e2e test for Wasm runtime telemetry v2 so wondering why this was not caught there.

**Comment by richardwxn** [comment]
it was caught in the perf pipeline and I am able to reproduce it locally

**Comment by bianpengyuan** [comment]
I cannot reproduce this with master build and Wasm runtime enablement with `--set values.telemetry.v2.metadataExchange.wasmEnabled=true --set values.telemetry.v2.prometheus.wasmEnabled=true`. Does perf pipeline uses the same option?

**Comment by richardwxn** [comment]
yes, and this is the config file: https://github.com/istio/tools/blob/master/perf/benchmark/configs/istio/telemetryv2_statswasm/installation.yaml. When I tested locally I just used these two set flags

**Comment by bianpengyuan** [comment]
Is this pipeline new or has it been running for a while? Did it start to fail recently?

**Comment by richardwxn** [comment]
Can you try this one locally: https://storage.googleapis.com/istio-build/dev/1.7-alpha.53348fe5830ce1ecd53ca85ee4f2ac9743490385/istio-1.7-alpha.53348fe5830ce1ecd53ca85ee4f2ac9743490385-linux-amd64.tar.gz

**Comment by bianpengyuan** [comment]
Still works fine for me.

**Comment by carolynhu** [comment]
Same issue happens on yesterday's perf pipeline for master, logs is here: https://storage.googleapis.com/istio-prow/logs/daily-fortio-performance-benchmark/20/build-log.txt

And 1.6-lastest has adds on installation related issues too, logs: https://storage.googleapis.com/istio-prow/logs/daily-nighthawk-performance-benchmark-release-1.6/20/build-log.txt

```
✘ Addons encountered an error: failed to wait for resource: resources not ready after 5m0s: timed out waiting for the condition
Deployment/istio-system/prometheus
- Pruning removed 
```
Not sure if they are related or not, but it worthwhile to take a look.

**Comment by bianpengyuan** [comment]
So this pipeline has never ran successfully, or did it start to fail recently?

**Comment by carolynhu** [comment]
It runs well until I enabled wasm run. Which is to set telemetryv2_statswasm=true in the perf.conf file. So I'm disabling it right now.

**Comment by bianpengyuan** [comment]
Enabling Wasm runtime will require higher CPU & memory at proxy loading/bootstrapping time. Readiness probe could fail because lack of resource. Do you know what is the resource provisioned on the testing node? And does client and server share the same test node?

**Comment by carolynhu** [comment]
The resource we provisioned for proxy is: 

```
sidecar.istio.io/proxyCPU: 1000m
sidecar.istio.io/proxyMemory: 512Mi
```

and the resource we requested for deployment is:

```
resources:
          requests:
            cpu: 800m
            memory: 1000Mi
```
Which I think I should increase the requested CPU?

Yes, client and server pods share the same test node.

**Comment by bianpengyuan** [comment]
How many cpu and memory do we provision for the test node? I tried the resource annotations and still works fine for me. So I think it might be that the test node does not enough cpu for Wasm module loading.

**Comment by carolynhu** [comment]
3 nodes, each node has 32vCPUs, 120G memory

**Comment by kyessenov** [comment]
How many workers do you have? 32 vCPU might cause auto detection to create 32 Wasm VMs.

**Comment by bianpengyuan** [comment]
The default installation have vm_id and should not trigger per thread VM creation. and I am pretty sure the work thread in perf test is 2. We recently just fixed a regression which was caused by worker number: https://github.com/istio/istio/issues/22845, but Carolyn could double check for sure.

**Comment by carolynhu** [comment]
yes, this issues comes from the latest master, which should be after the above fix.

**Comment by richardwxn** [comment]
It is fixed in master/release-1.7 already
